### DISCUSSION TITLE
Copying SID from table/data grid

### DISCUSSION THREAD

**Discussion by Khushli** [twilio-labs/paste#2160] [General] [answered]
Hi, 

I have a pretty long Connection SID number (34 characters) to show in a table and was looking for suggestions on truncating it while having the ability to copy the SID. 

I believe this has been done in the past but I can't seem to find examples. Looking for examples and suggestions.

**Comment by PoojMenon** [comment, marked as answer]
We did for Monkey. Here is a link to the design. 
https://www.figma.com/file/XBsuXuvbILfuTpfDon1Zb1/2021-Q3---Reviewing-usage-in-Monkey?node-id=793%3A102254

**Reply by Khushli** [reply]
@PoojMenon I see the truncated SID but what is the interaction to copy it? Couldn't find that in the file. Can you point me to it? Thanks

**Reply by PoojMenon** [reply]
We didn't actually have a need to copy that particular SID, so it isn't present on that particular screen. We have a slightly different use case in the screen below where we display the friendly name instead of SID and the copy interaction is within a popover.
https://www.figma.com/file/XBsuXuvbILfuTpfDon1Zb1/2021-Q3---Reviewing-usage-in-Monkey?node-id=1419%3A245221

**Reply by Khushli** [reply]
That popover makes sense but it needs a clickable button to activate it. I already have a anchor in the table to link out to the connection details page. I don't know the guidance on having two anchor buttons next to each other in the same table.
<img width="976" alt="Screen Shot 2022-01-26 at 4 12 17 PM" src="https://user-images.githubusercontent.com/87436655/151259876-5c56d07d-a61d-4631-a724-bd81857d0069.png">

**Reply by shleewhite** [reply]
Hi @Khushli! Just to clarify a little bit, do you mean having a button on the right of the SID?

**Reply by Khushli** [reply]
The [connection name] is a anchor button. To make a popover, [connection SID] becomes needs to be button too, which I think could be an issue since I want to maintain the hierarchy of the [connection name] button which links to the details page for that connection. 

I just need a way to enable copying the SID without having the full SID on the table because it's simply to long to fit.

**Reply by shleewhite** [reply]
Totally. I don't think having multiple interactive elements on a row would necessarily disrupt the hierarchy, but would having a copy button in the cell work? It could be a button like the example @PoojMenon shared

<img width="249" alt="Screen Shot 2022-01-27 at 2 57 34 PM" src="https://user-images.githubusercontent.com/8482334/151457223-d4a0852f-fbd0-4757-9a34-726ac7ae09bb.png">

**Reply by Khushli** [reply]
Yes, that could work. Thanks!

**Reply by Khushli** [reply]
<img width="850" alt="Screen Shot 2022-01-28 at 6 11 29 PM" src="https://user-images.githubusercontent.com/87436655/151634162-c38f44d8-d11b-453b-bff3-1f92a6e2ed42.png">

This is how it looks... let me know if this looks okay

**Reply by shleewhite** [reply]
Looks great! A note for implementation: we recommend using [the Icon Button component](https://paste.twilio.design/components/button/#icon-buttons) and having an affordance to let the user know that it copied successfully. It could be a tooltip, like in this example: 
![image](https://user-images.githubusercontent.com/8482334/151862368-0154017b-41bd-4763-9c24-262886e52b4e.png)
